Infrastructure/Services/ProcessEmailsService.cs

```diff
@@ -0,0 +1,450 @@
+using EmailStatus = Domain.Enums.EmailStatus;
+
+namespace Infrastructure.Services;
+
+public class ProcessEmailsService : IProcessEmailsService
+{
+    private const string ScheduleName = "Process Emails";
+    private const string TestExtraInfoKey = "test_extra_info";
+
+    private readonly IUnitOfWork _unitOfWork;
+    private readonly ILogger<AddEnquiryCommandHandler> _logger;
+    private readonly INotificationsClientService _notificationsClientService;
+    private readonly EmailSettings _emailSettingsConfig;
+
+    public ProcessEmailsService(IUnitOfWork unitOfWork,
+        INotificationsClientService notificationsClientService,
+        ILogger<AddEnquiryCommandHandler> logger,
+        IOptions<EmailSettings> emailSettingsConfig)
+    {
+        _unitOfWork = unitOfWork;
+        _notificationsClientService = notificationsClientService;
+        _logger = logger;
+        _emailSettingsConfig = emailSettingsConfig.Value;
+    }
+
+    public async Task<ProcessedEmailsModel> ProcessAllEmailsAsync()
+    {
+        var processedEmailsModel = new ProcessedEmailsModel();
+        var startDate = DateTime.UtcNow;
+        _logger.LogInformation("Starting ProcessAllEmailsAsync at {startDate}", startDate);
+
+        try
+        {
+            var startedOK = await StartProcessing(startDate);
+            if (startedOK)
+            {
+                if (_emailSettingsConfig.SendEmailsFromNtp)
+                {
+                    await PollForStatusUpdatesAsync(processedEmailsModel);
+
+                    processedEmailsModel.EmailsSent = await SendEmailsToBeProcessedAsync(null, false);
+
+                    var finishedDate = DateTime.UtcNow;
+                    var finishedMessage = $"Completed email process successfully.  Started at {startDate}, finished at {finishedDate}, taking {(finishedDate - startDate).TotalSeconds} seconds.  " +
+                        $"{processedEmailsModel.EmailsCheckStatus} email status checks, {processedEmailsModel.EmailsUpdatedStatus} changed email status, {processedEmailsModel.EmailsSent} emails sent.";
+
+                    await FinishProcessing(finishedDate, finishedMessage);
+
+                    processedEmailsModel.Outcome = finishedMessage;
+                }
+                else
+                {
+                    await SetEmailsToDeactivatedEmailsAsync();
+                    processedEmailsModel.Outcome = "Sending of emails is switched off in the config";
+                    await FinishProcessing(DateTime.UtcNow, processedEmailsModel.Outcome);
+                }
+            }
+            else
+            {
+                processedEmailsModel.Outcome = $"Previous email process is still running at {startDate}.";
+            }
+
+            _logger.LogInformation(processedEmailsModel.Outcome);
+        }
+        catch (Exception ex)
+        {
+            await FinishProcessing(DateTime.UtcNow, $"Unexpected exception thrown. {ex}");
+            throw;
+        };
```

Comment:
Empty statements, i.e. ;, should be removed.

Reply:
Cheers, hadn't spotted that one.  Done